### DISCUSSION TITLE
got srsran 5g working with Benetel 650A

### DISCUSSION THREAD

**Discussion by meninob** [srsran/srsRAN_Project#744] [Ideas & Suggestions]
Hi all,
With the help of Andre(SRS) and another vendor(Fibrolan = Idan), I managed to get the 5G working using srsRAN_Project, with Open5Gs, along with a Fibrolan PTP switch, Benetel 650A radio, (and even Ettus B210 radio device), along with Dlink DWP 1010 with M15 WAP access point for users.
I just got laid off from the company after deploying it, but the documentation is in their repo, but the information is still in my head. 
I also have the scripts and the configuration, if anyone needs it, let me know, and I will post it, or help you where and when permissible.
Note that I used open5gs All-in-one mode, and was going to get DPDK working, as my next step, but did not get the chance.
I think it can work with another switch which has PTP, but when tested with Cisco NCS540, the PTP readouts were ok (below 100ms), but the phc sync had about 37seconds offset, and hence did not work properly with this switch, and might have to do with the gps sync, or the ptp / p2p config itself, but I can't explore it further now.
Anyways, if anyone needs help on this, let me know, and will provide what I can, while its still in memory.

**Comment by meninob** [comment]
Attaching the configuration I used for SRSRAN_project, which gave me quite good throughput at almost 700mbps.
you will notice a lot of comments in the config, for values, as I was testing and tweaking it (still will need tweaking, I guess, but that is now your prerogative).
I increased the gtpu queue size, and I got better results.
Not everything is commented, and I copied off variables from different settings I found off of SRS github and from SRS themselves, so use freely.
Hope this helps someone
[ogstun.sh.txt](https://github.com/user-attachments/files/16441901/ogstun.sh.txt)
[benetel_radio_init.sh.txt](https://github.com/user-attachments/files/16441902/benetel_radio_init.sh.txt)

[ptp_phc.cfg.txt](https://github.com/user-attachments/files/16441894/ptp_phc.cfg.txt)

[gnb_ru_ran650_100mhz.yml.txt](https://github.com/user-attachments/files/16441858/gnb_ru_ran650_100mhz.yml.txt)

Also attaching the scripts I used, and the ptp config, as well as the ogstun initialization script for UE internet access.
Hope this helps someone.

Let me know if you need more information. Will be happy to share what I learnt.
Thanks to @AhmedBatra for feedback on being able to use DPDK, using 2 interfaces.... one for DPDK and the other for ptp sync. I wanted to try that, and glad he's done it and confirmed it works. kudos to you on this, Ahmed Batra.

**Comment by pgawlowicz** [comment]
Nice, thanks for your feedback!
